smartclide-monitoring/src/main/java/eu/smartclide/contexthandling/services/GitRestCallService.java

```diff
@@ -0,0 +1,183 @@
+                    gitMessage.setNoOfModifiedFiles(newCommitDiff.size());
+
+                    gitMessages.add(gitMessage);
+                }
+            }
+        }
+        return gitMessages;
+    }
+
+    public Integer calculateTimeSinceLastCommit(String projectId, JsonObject newCommit) {
+        int difference = 0;
+        String lastCommitId = newCommit.get("parent_ids").getAsString();
+        String newCommitCreationDateStr = newCommit.get("created_at").getAsString();
+        JsonObject lastCommit = getCommitById(projectId, lastCommitId);
+        String lastCommitCreationDateStr = lastCommit.getAsJsonObject().get("created_at").getAsString();
+
+        DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSXXX");
+        try {
+            Date newCommitCreationDate = formatter.parse(newCommitCreationDateStr);
+            Date lastCommitCreationDate = formatter.parse(lastCommitCreationDateStr);
+            difference = Math.toIntExact(Math.abs(newCommitCreationDate.getTime() - lastCommitCreationDate.getTime()) / 1000);
+        } catch (java.text.ParseException e) {
+            logger.error("date to string parse error", e);
+        }
+        return difference;
+    }
+
+    public JsonArray getAllBranchesForGivenProject(String projectId) {
+        return parseHttpResponseToJsonArray(makeGetCallToGitlab(baseUri + projectId + uriPartForBranches + uriParams));
+    }
+
+    public JsonObject getCommitById(String projectId, String commitId) {
+        return parseHttpResponseToJsonObject(makeGetCallToGitlab(baseUri + projectId
+                + uriPartForCommits + commitId + uriParams));
+    }
+
+    public JsonArray getCommitDiff(String projectId, String commitId) {
+        return parseHttpResponseToJsonArray(makeGetCallToGitlab(baseUri + projectId
+                + uriPartForCommits + commitId + uriPartForDiff + uriParams));
+    }
+
+    public JsonArray getCommitsForGivenBranchAndSince(String projectId, String branchName, String sinceParam) {
+        return parseHttpResponseToJsonArray(makeGetCallToGitlab(baseUri + projectId
+                + uriPartForCommits + uriParams + refNameParam + branchName + sinceParam));
+    }
+
+    public JsonArray getUserProjects() {
+        return parseHttpResponseToJsonArray(makeGetCallToGitlab(baseUri + uriParams));
+    }
+
+    /**
+     * this method makes get call to gitlab server with given uri
+     *
+     * @param uri as string
+     * @return HttpResponse<String> as response
+     */
+    private HttpResponse<String> makeGetCallToGitlab(String uri) {
+
+        final HttpClient httpClient = HttpClient.newBuilder()
+                .version(HttpClient.Version.HTTP_2)
+                .connectTimeout(Duration.ofSeconds(5 * 60))
+                .build();
+
+        HttpRequest request = HttpRequest.newBuilder().GET().uri(URI.create(uri)).build();
+
+        HttpResponse<String> response = null;
+        try {
```

Comment:
I suggest to move the `try` to the beginning of the method, and thus enclose the whole method body into the `try-catch`.